samples/vision_inference/README.md

```diff
@@ -0,0 +1,29 @@
+# Vision Inference Sample
+
+This sample demonstrates how to run a MNIST handwritten digit detection vision
+model on an image using IREE's command line tools.
+
+## Instructions
+
+From this directory:
+
+```bash
+# Compile the MNIST program.
+iree-compile \
+    ../models/mnist.mlir \
+    --iree-input-type=mhlo \
+    --iree-hal-target-backends=cpu \
+    -o /tmp/mnist_cpu.vmfb
+
+# Convert the test image to the 1x28x28x1xf32 buffer format the program expects.
+cat mnist_test.png | python3 convert_to_float_grayscale.py > /tmp/mnist_test.bin
+
+# Run the program, passing the path to the binary file as a function input.
+iree-run-module \
+  /tmp/mnist_test.bin \
+  --function_input=1x28x28x1xf32=@/tmp/mnist_test.bin
+
+# Observe the results - a list of prediction confidence scores for each digit.
+```
+
+<!-- TODO(scotttodd): lit test for that ^ (requires python in lit.cfg.py) -->
```

Comment:
I spent a good 30 minutes trying to figure out how to copy the magic from
https://github.com/google/iree/blob/74d741e62c0646ddd34a927ace594b8829858d48/llvm-external-projects/iree-dialects/test/lit.cfg.py#L65-L70 https://github.com/google/iree/blob/74d741e62c0646ddd34a927ace594b8829858d48/llvm-external-projects/iree-dialects/test/CMakeLists.txt#L5-L10 https://github.com/google/iree/blob/74d741e62c0646ddd34a927ace594b8829858d48/llvm-external-projects/iree-dialects/test/lit.site.cfg.py.in#L15
into https://github.com/google/iree/blob/main/samples/lit.cfg.py, but had no luck (`llvm_config` was always `None`). Could be Windows / Python making it complicated... idk.

Reply:
`llvm_config` is from using llvm-lit, which isn't the same as lit, which is... cool

Reply:
Ah, so would we need to swap out `${LLVM_SOURCE_DIR}/utils/lit/lit.py` in `iree_lit_test.cmake`? https://github.com/google/iree/blob/d97e8bc0d306568ea047f408610779f7450f9230/build_tools/cmake/iree_lit_test.cmake#L67-L78
Could we just always use `llvm-lit`?

Reply:
llvm-lit is weird and I don't really understand it. You're welcome to look into this, but it's a whole can of worms (don't forget about Bazel)

Reply:
I just want to run a python script in a test 😭 (well, `iree-compile` -> python script -> `iree-run-module`). I'll punt on test coverage for now I guess...